<a href="https://colab.research.google.com/github/abhinandan6123/AI-Projects/blob/main/Observations_and_Insights.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:

# Step 1: Import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns



# Step 2: Load the dataset
# Upload your file manually when prompted by Colab
from google.colab import files
uploaded = files.upload()





Saving processed_logistics_data (2).csv to processed_logistics_data (2).csv


In [4]:
# # Loading the uploaded CSV file to understand its structure and content for analysis
import pandas as pd
import io

# Access the uploaded file from the 'uploaded' dictionary
# Assuming 'processed_logistics_data (2).csv' is the filename in the dictionary
file_name = 'processed_logistics_data (2).csv'  # Replace with the actual file name if different
data = pd.read_csv(io.BytesIO(uploaded[file_name]))

# Display basic information about the data
data_info = {
    "columns": data.columns.tolist(),
    "num_rows": data.shape[0],
    "num_columns": data.shape[1],
    "sample_data": data.head()
}
data_info


{'columns': ['data',
  'trip_creation_time',
  'route_schedule_uuid',
  'route_type',
  'trip_uuid',
  'source_center',
  'source_name',
  'destination_center',
  'destination_name',
  'od_start_time',
  'od_end_time',
  'start_scan_to_end_scan',
  'is_cutoff',
  'cutoff_factor',
  'cutoff_timestamp',
  'actual_distance_to_destination',
  'actual_time',
  'osrm_time',
  'osrm_distance',
  'factor',
  'segment_actual_time',
  'segment_osrm_time',
  'segment_osrm_distance',
  'segment_factor',
  'trip_duration',
  'trip_creation_day',
  'trip_creation_month'],
 'num_rows': 227,
 'num_columns': 27,
 'sample_data':        data       trip_creation_time  \
 0  training  2018-09-20 02:35:36.477   
 1  training  2018-09-20 02:35:36.477   
 2  training  2018-09-20 02:35:36.477   
 3  training  2018-09-20 02:35:36.477   
 4  training  2018-09-20 02:35:36.477   
 
                                  route_schedule_uuid route_type  \
 0  thanos::sroute:eb7bfc78-b351-4c0e-a951-fa3d5c3...    Carting  

The dataset contains 27 columns and 227 rows, with features that include both numerical and categorical variables.

**Here's a brief analysis of how you might approach each type of observation:**

**Univariate Observations**


**1. Numerical Features:**

Analyze which features have a normal distribution, are skewed, or contain outliers. This can be done using histograms or summary statistics.

**Example numerical features:**

actual_distance_to_destination, actual_time, osrm_time, trip_duration, etc.

**Categorical Features:**

Determine dominant or evenly distributed categories for features like route_type, source_center, and destination_center.

Frequency counts or bar plots can highlight which categories are dominant.

**Bivariate Observations**

**1. Numerical Pairs:**

Identify high correlations and trends between pairs of numerical features, such as actual_time vs. osrm_time or actual_distance_to_destination vs. osrm_distance.

A heatmap or scatter plots are useful here.

**2. Categorical vs. Numerical:**

Analyze how numerical features vary across different categories. For example, differences in trip_duration based on route_type can reveal significant patterns.

Box plots or violin plots can be helpful for this comparison.

**Multivariate Observations**

**1. PCA Plot for Clustering:**

Conduct PCA on the numerical features to reduce dimensionality and visualize if any clusters are formed, indicating natural groupings within the data.

**2.Parallel Coordinates:**


Use parallel coordinates to explore patterns across multiple dimensions, focusing on specific categories to see if there are consistent patterns across features.